test/built-ins/TypedArray/prototype/toString/this-is-not-object.js

```diff
@@ -0,0 +1,54 @@
+// Copyright (C) 2016 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-%typedarray%.prototype.toString
+description: Throws a TypeError exception when `this` is not Object
+info: >
+  22.2.3.29 %TypedArray%.prototype.toString ( )
+
+  ...
+
+  22.2.3.15 %TypedArray%.prototype.join ( separator )
+
+  This function is not generic. ValidateTypedArray is applied to the this value
+  prior to evaluating the algorithm. If its result is an abrupt completion that
+  exception is thrown instead of evaluating the algorithm.
+
+  22.2.3.5.1 Runtime Semantics: ValidateTypedArray ( O )
+
+  1. If Type(O) is not Object, throw a TypeError exception.
```

Comment:
there's no validation on toString before calling join. 

The full description on toString omits the validation:

```

22.2.3.29 %TypedArray%.prototype.toString ( )

The initial value of the %TypedArray%.prototype.toString data property is the same built-in function object as the Array.prototype.toString method defined in 22.1.3.28.
```

Then Array.prototype.toString will be called first, getting the `join` method:

```
22.1.3.28 Array.prototype.toString ( )

When the toString method is called, the following steps are taken:

    1. Let array be ? ToObject(this value).
    2. Let func be ? Get(array, "join").
    3. If IsCallable(func) is false, let func be the intrinsic function %ObjProto_toString%.
    4. Return ? Call(func, array). 
```